### PR TITLE
feat: Add onAny and onError functions

### LINKED ISSUE #304: Create .onAny and .onError methods, deprecate "*" and "error" events
Follow up to https://github.com/octokit/webhooks.js/issues/180

I would reserve the `.on(event, handler)` method for actual events only, and define new methods for special events that are unique to `@octokit/webhooks`.

That will also address https://github.com/octokit/webhooks.js/issues/302, because we would no longer need to define `*` and `error` as part of the overall event map

### PR DESCRIPTION
**Pull request by wolfy1339**
Fixes #304

### CODE REVIEW

**Review by oscard0m — DISMISSED** [review]
LGTM

File: src/event-handler/index.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export function createEventHandler(options: Options<any>) {
 
   return {
     on: on.bind(null, state),
+    onAny: on.bind(null, state).bind(null, "*"),
+    onError: on.bind(null, state).bind(null, "error"),
```
Comment (wolfy1339) [inline src/event-handler/index.ts]: This is probably not the best solution. But it ensures backwards compatibility and it works

File: test/typescript-validate.ts [hunk 2]
```diff
@@ -69,12 +69,19 @@ export default async function () {
 
   verify("randomSecret", {}, "randomSignature");
 
+  // This is deprecated usage
```
Comment (oscard0m) [inline test/typescript-validate.ts:72]: 💡  there's a deprecated section at the end of this file :)

File: test/typescript-validate.ts [hunk 3]
```diff
@@ -124,6 +131,7 @@ export default async function () {
     console.log(event.foo);
   });
 
+  // This is deprecated usage
```
Comment (oscard0m) [inline test/typescript-validate.ts:134]: https://github.com/octokit/webhooks.js/pull/324/files#r504256440

File: src/event-handler/index.ts (hunk 1 above)
Comment (gr2m) [reply src/event-handler/index.ts]: I would want to log a deprecation message when `.on()` is called with either `*` or `error`. While this is a neat trick, it won't allow us to do that.

I'd also like to add tests for `.on("error", handler)` and `.on("*", handler)` to a new `test/integration/deprecations-test.js` file, similar to what we have in other @octokit repositories. That makes it easier to prepare new breaking releases, and having these tests prevents accidental regressions. The tests should test for the deprecation waring to be logged

Comment (gr2m) [reply src/event-handler/index.ts]: btw you can bind as many arguments as you like, so you could do this instead: 

```js
    onAny: on.bind(null, state, "*"),
    onError: on.bind(null, state, "error"),
```

Comment (wolfy1339) [reply src/event-handler/index.ts]: Done

**Comment by wolfy1339** [comment]
@gr2m I can't seem to push to the branch anymore

**Comment by gr2m** [comment]
@wolfy1339 sorry for that. I've invited you and @dominguezcelada to the new @octokit/js-community team, you should have write access to all JS repositories now :)

**Review by gr2m — APPROVED** [review]
looks great 👍🏼

**Review by oscard0m — APPROVED** [review]

**Pull request merged**